src/main/java/org/spongepowered/common/data/processor/data/entity/FuseDataProcessor.java

```diff
@@ -0,0 +1,133 @@
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableFuseData;
+import org.spongepowered.api.data.manipulator.mutable.entity.FuseData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.MutableBoundedValue;
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeFuseData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeFuseData;
+import org.spongepowered.common.data.processor.common.AbstractEntitySingleDataProcessor;
+import org.spongepowered.common.data.processor.common.AbstractSingleDataProcessor;
+import org.spongepowered.common.data.value.SpongeValueFactory;
+
+import java.util.Optional;
+
+public class FuseDataProcessor extends AbstractEntitySingleDataProcessor<Entity, Integer, MutableBoundedValue<Integer>, FuseData, ImmutableFuseData> {
+
+    public FuseDataProcessor() {
+        super(Entity.class, Keys.FUSE_DURATION);
+    }
+
+    @Override
+    protected FuseData createManipulator() {
+        return new SpongeFuseData();
+    }
+
+    @Override
+    public DataTransactionResult removeFrom(ValueContainer<?> container) {
+        return DataTransactionResult.failNoData();
+    }
+
+    @Override
+    protected MutableBoundedValue<Integer> constructValue(Integer actualValue) {
+        return SpongeValueFactory.boundedBuilder(Keys.FUSE_DURATION)
+                .minimum(-1)
+                .maximum(Integer.MAX_VALUE)
+                .defaultValue(0)
+                .actualValue(actualValue)
+                .build();
+    }
+
+    @Override
+    protected boolean set(Entity container, Integer value) {
+        if (this.supports(container)) {
+            if (container instanceof EntityCreeper) {
+                ((EntityCreeper) container).fuseTime = value;
+            } else if (container instanceof EntityFireworkRocket) {
+                ((EntityFireworkRocket) container).lifetime = value;
+            } else if (container instanceof EntityMinecartTNT) {
+                ((EntityMinecartTNT) container).minecartTNTFuse = value;
+            } else if (container instanceof EntityTNTPrimed) {
+                ((EntityTNTPrimed) container).fuse = value;
+            }
+            
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    protected Optional<Integer> getVal(Entity container) {
+        if (container instanceof EntityCreeper) {
+            EntityCreeper creeper = (EntityCreeper) container;
+            
+            return Optional.of(creeper.fuseTime);
```

Comment:
This isn't correct, as `timeSinceIgnited` counts *up* to the fuse values for creepers. With the current code, creepers will always show the same fuse value, even while ignited.